starlette/background.py

```diff
@@ -41,5 +41,7 @@ def add_task(
         self.tasks.append(task)
 
     async def __call__(self) -> None:
-        for task in self.tasks:
+        for task in self.tasks.copy():
             await task()
+            if task in self.tasks:
+                self.tasks.remove(task)
```

Comment:
Why would the task ever not be in tasks?

Reply:
The reason, for this is because once the task is processed in `await task()`, why is there a need to keep it in the ram. I am removing it because, I want to achieve something like this

If I want to fetch the current task queue length

using something like below

```python
class QueueOut(BaseModel):
    task_lenth: int

@app.get("/get_queue_length/", response_model=QueueOut)
async def read_task():
    return QueueOut(task_lenth=len(tasks.tasks))
    
  ```
This is helpful for graceful shutdown of the application in scenarios where a lots of bg task are enqueued and now you intercepted a SIGTERM or SIGKILL which sends a termination signal but, the server will not be stopped until all the enqueued tasks are processed.